control/mpc_lateral_controller/src/mpc_lateral_controller.cpp

```diff
@@ -227,6 +231,28 @@ std::shared_ptr<SteeringOffsetEstimator> MpcLateralController::createSteerOffset
   return steering_offset_;
 }
 
+void MpcLateralController::setStatus(
+  diagnostic_updater::DiagnosticStatusWrapper & stat)
+{
+  if (m_is_mpc_solved) {
+    stat.summary(diagnostic_msgs::msg::DiagnosticStatus::OK, "MPC succeeded.");
+  } else {
+    const std::string & error_msg = "The MPC solver failed. Call MRM to stop the car.";
```

Comment:
You can not use `&` for a temporal object. This code may result in undefined-behavior.

```suggestion
    const std::string error_msg = "The MPC solver failed. Call MRM to stop the car.";
```